File: src/main/java/com/example/MonaServer/Helper/MyUserDetailsService.java
package com.example.MonaServer.Helper;


import com.example.MonaServer.Entities.User;
import com.example.MonaServer.Repository.UserRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Component;

import java.util.Collections;
import java.util.Optional;

@Component
public class MyUserDetailsService implements UserDetailsService {

    @Autowired private UserRepo userRepo;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        Optional<User> userRes = userRepo.findById(username);
        if(userRes.isEmpty())
            throw new UsernameNotFoundException("Could not findUser with username = " + username);
        User user = userRes.get();
        return new org.springframework.security.core.userdetails.User(
                username,
                user.getPassword(),
                Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER")));
    }
}

File: src/main/kotlin/de/lrprojects/monaserver/security/MyUserDetailsService.kt
package de.lrprojects.monaserver.security

import de.lrprojects.monaserver.entity.User
import de.lrprojects.monaserver.properties.RoleConstants.ADMIN_ROLE
import de.lrprojects.monaserver.properties.RoleConstants.USER_ROLE
import de.lrprojects.monaserver.properties.TokenProperties
import de.lrprojects.monaserver.repository.UserRepository
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import java.util.*


class MyUserDetailsService(
    private val userRepository: UserRepository,
    private val tokenProperties: TokenProperties
) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        val userRes: Optional<User> = userRepository.findById(UUID.fromString(username))
        if (userRes.isEmpty) throw UsernameNotFoundException("Could not findUser with username = $username")
        val user: User = userRes.get()
        val roles = mutableListOf(SimpleGrantedAuthority(USER_ROLE))
        if (user.username == tokenProperties.adminAccountName) {
            roles.add(SimpleGrantedAuthority(ADMIN_ROLE))
        }
        return org.springframework.security.core.userdetails.User(
            user.id.toString(), user.password, roles
        )
    }
}